data-prepper-plugins/otlp-source/src/main/java/org/opensearch/dataprepper/plugins/source/otlp/OTLPSource.java

```diff
@@ -0,0 +1,313 @@
+  OTLPSource(final OTLPSourceConfig otlpSourceConfig,
+      final PluginMetrics pluginMetrics,
+      final PluginFactory pluginFactory,
+      final CertificateProviderFactory certificateProviderFactory,
+      final PipelineDescription pipelineDescription) {
+    otlpSourceConfig.validateAndInitializeCertAndKeyFileInS3();
+    this.otlpSourceConfig = otlpSourceConfig;
+    this.pluginMetrics = pluginMetrics;
+    this.certificateProviderFactory = certificateProviderFactory;
+    this.pipelineName = pipelineDescription.getPipelineName();
+    this.authenticationProvider = createAuthenticationProvider(pluginFactory);
+  }
+
+  @Override
+  public void start(Buffer<Record<Object>> buffer) {
+    if (buffer == null) {
+      throw new IllegalStateException("Buffer provided is null");
+    }
+
+    if (server == null) {
+      @SuppressWarnings("unchecked")
+      Buffer<Record<? extends Metric>> metricBuffer = (Buffer<Record<? extends Metric>>) (Object) buffer;
+
+      final OTelLogsGrpcService oTelLogsGrpcService = new OTelLogsGrpcService(
+          (int) (otlpSourceConfig.getRequestTimeoutInMillis() * 0.8),
+          new OTelProtoStandardCodec.OTelProtoDecoder(),
+          buffer, pluginMetrics);
+
+      final OTelMetricsGrpcService oTelMetricsGrpcService = new OTelMetricsGrpcService(
+          (int) (otlpSourceConfig.getRequestTimeoutInMillis() * 0.8),
+          new OTelProtoStandardCodec.OTelProtoDecoder(),
+          metricBuffer, pluginMetrics);
+
+      final OTelTraceGrpcService oTelTraceGrpcService = new OTelTraceGrpcService(
+          (int) (otlpSourceConfig.getRequestTimeoutInMillis() * 0.8),
+          new OTelProtoStandardCodec.OTelProtoDecoder(),
+          buffer, pluginMetrics);
+
+      final List<ServerInterceptor> serverInterceptors = getAuthenticationInterceptor();
+
+      final GrpcServiceBuilder grpcServiceBuilder = GrpcService
+          .builder()
+          .useClientTimeoutHeader(false)
+          .useBlockingTaskExecutor(true)
+          .exceptionHandler(createGrpExceptionHandler());
+
+      final MethodDescriptor<ExportLogsServiceRequest, ExportLogsServiceResponse> methodLogsDescriptor = LogsServiceGrpc
+          .getExportMethod();
+      final MethodDescriptor<ExportMetricsServiceRequest, ExportMetricsServiceResponse> methodMetricsDescriptor = MetricsServiceGrpc
+          .getExportMethod();
+      final MethodDescriptor<ExportTraceServiceRequest, ExportTraceServiceResponse> methodTraceDescriptor = TraceServiceGrpc
+          .getExportMethod();
+
+      final String oTelLogsSourcePath = otlpSourceConfig.getLogsPath();
+      final String oTelMetricsSourcePath = otlpSourceConfig.getMetricsPath();
+      final String oTelTraceSourcePath = otlpSourceConfig.getTracesPath();
+
+      // If the path is null for any of the sources, we will not transform the path
+      // and use the default path
+      if (oTelTraceSourcePath != null && oTelMetricsSourcePath != null && oTelLogsSourcePath != null) {
```

Comment:
Why is this check not performed separately on each source path?

Reply:
@KarstenSchnitter This actually made more sense, once I looked into this further. I have changed my approach to accommodate this 😄